snippets/elo.md

```diff
@@ -15,17 +15,20 @@ const elo = ([...ratings], kFactor = 32, selfRating) => {
   const expectedScore = (self, opponent) => 1 / (1 + 10 ** ((opponent - self) / 400));
   const newRating = (rating, i) =>
     (selfRating || rating) + kFactor * (i - expectedScore(i ? a : b, i ? b : a));
+    
   if (ratings.length === 2) {
     return [newRating(a, 1), newRating(b, 0)];
-  } else {
-    for (let i = 0; i < ratings.length; i++) {
-      let j = i;
-      while (j < ratings.length - 1) {
-        [ratings[i], ratings[j + 1]] = elo([ratings[i], ratings[j + 1]], kFactor);
-        j++;
-      }
+  } 
+  
+  for (let i = 0, l = ratings.length; i < l; i++) {
```

Comment:
While it leads to a speed improvements I think we do to much short cutting of names into abstracts here. We shoudl probably expand the names to make this more readable than a bunch of single letters

Reply:
`i` and `j` as iterators are fine (they are also part of the guidelines). `len` or `length` for the length should be a bit better than the current name.